fundingmanager_test.go

```diff
@@ -536,22 +536,19 @@ func openChannel(t *testing.T, alice, bob *testNode, localFundingAmt,
 
 func assertNumPendingChannels(t *testing.T, node *testNode, expectedNum int) {
 	var numPendingChans int
-	for i := 0; i < testPollNumTries; i++ {
-		// If this is not the first try, sleep before retrying.
-		if i > 0 {
-			time.Sleep(testPollSleepMs * time.Millisecond)
-		}
-		pendingChannels, err := node.fundingMgr.
-			cfg.Wallet.Cfg.Database.FetchPendingChannels()
-		if err != nil {
-			t.Fatalf("unable to fetch pending channels: %v", err)
-		}
 
-		numPendingChans = len(pendingChannels)
-		if numPendingChans == expectedNum {
-			// Success, return.
-			return
-		}
+	time.Sleep(testPollSleepMs * time.Millisecond * testPollNumTries)
```

Comment:
I see why that was changed. However, for the test in my PR it's going to return a false positive when the channel does get closed. 

How does the previous implementation solve the issues with false negatives? If the action does not occur within X seconds, how does checking N times within the interval help?

Reply:
Do the false positives you're observing occur on the last call to `assertNumPendingChans`?

Reply:
They occur when we are testing an action should not have happened.
For example if we test a channel close shouldn't have occurred.

So if our original channel count is 1, and we assert the count is 1.
Then the first iteration of the loop will return true, even if a channel is deleted on loop 2.

Reply:
Ahh okay, I see. Unfortunately, this is a common headache of testing asynchronous interfaces. 

I would suggest making a separate method to assert this property since it is different from what `assertNumChannels` is testing. The original method is actually testing `assertNumChannelsBecomes(x)` within some timeout, where as your test requires something like `assertNumChannelsRemains(x)` for some equivalent timeout. Both can poll and break out early if the predicate is satisfied in the first case, or the invariant is violated in the second.